test/extended/etcd/helpers/helpers.go

```diff
@@ -597,6 +600,114 @@ func MachineNameToEtcdMemberName(ctx context.Context, kubeClient kubernetes.Inte
+	}
+
+	// case 2: match on an internal ip address
+	nodeIPListSet := sets.NewString()
+	for _, addr := range node.Status.Addresses {
+		if addr.Type == corev1.NodeInternalIP {
+			nodeIPListSet.Insert(addr.Address)
+		}
+	}
+
+	var machineNames []string
+	for _, masterMachine := range masterMachines.Items {
+		for _, addr := range masterMachine.Status.Addresses {
+			if addr.Type == corev1.NodeInternalIP {
+				if nodeIPListSet.Has(addr.Address) {
+					return masterMachine.Name, nil
+				}
+			}
+			machineNames = append(machineNames, masterMachine.Name)
+		}
+	}
+
+	return "", fmt.Errorf("unable to find a machine for the correspoding %q node on the following nodes IP: %v, checked: %v", nodeName, nodeIPListSet.List(), machineNames)
+}
+
+// StopKubelet stops the kubelet on the given node by spwaning a pod on the node and running the stop kubelet command
+func StopKubelet(ctx context.Context, adminKubeClient kubernetes.Interface, node corev1.Node) error {
+	podSpec := applycorev1.PodSpec().WithRestartPolicy(corev1.RestartPolicyNever).WithHostNetwork(true).WithHostPID(true)
+	podSpec.Containers = []applycorev1.ContainerApplyConfiguration{
+		*applycorev1.Container().
+			WithName("kubelet-stopper").
+			WithSecurityContext(applycorev1.SecurityContext().WithPrivileged(true).WithRunAsUser(0)).
+			WithImage(image.ShellImage()).
+			WithVolumeMounts(applycorev1.VolumeMount().WithName("host").WithMountPath("/host")).
+			WithCommand("/bin/sh").
+			WithArgs("-c", "chroot /host /bin/sh -c 'sleep 1 && systemctl stop kubelet'"),
+	}
+	podSpec.NodeSelector = map[string]string{"kubernetes.io/hostname": node.Labels["kubernetes.io/hostname"]}
+	podSpec.Tolerations = []applycorev1.TolerationApplyConfiguration{*applycorev1.Toleration().WithOperator(corev1.TolerationOpExists)}
+	podSpec.Volumes = []applycorev1.VolumeApplyConfiguration{
+		*applycorev1.Volume().WithName("host").WithHostPath(applycorev1.HostPathVolumeSource().WithPath("/").WithType("Directory")),
+	}
+
+	pod := applycorev1.Pod("kubelet-stopper", "openshift-etcd").WithSpec(podSpec)
+	createdPod, err := adminKubeClient.CoreV1().Pods(*pod.Namespace).Apply(context.Background(), pod, metav1.ApplyOptions{FieldManager: *pod.Name})
+	if err != nil {
+		return fmt.Errorf("error applying pod %w", err)
+	}
+
+	err = wait.PollUntilContextTimeout(ctx, 15*time.Second, 5*time.Minute, true, func(ctx context.Context) (done bool, err error) {
+		p, err := adminKubeClient.CoreV1().Pods("openshift-etcd").Get(context.Background(), createdPod.Name, metav1.GetOptions{})
+		if err != nil {
+			return true, fmt.Errorf("error retreving the pod %s : %w", createdPod.Name, err)
+		}
+
+		if p.Status.Phase == corev1.PodFailed {
+			return true, fmt.Errorf("pod failed with status %v", p.Status)
+		}
+
+		podReadyCondition := e2epod.FindPodConditionByType(&p.Status, corev1.PodReady)
+		if podReadyCondition == nil {
+			return false, nil
+		}
+
+		return podReadyCondition.Status == corev1.ConditionFalse, nil
```

Comment:
you have to be a bit careful here, kubelet is the only one updating the status - if you shut it down this condition may never become true. I would just try to fire and forget this pod and wait for the node to become not ready.